data/lib/compat/compat.lua

```diff
@@ -580,12 +580,10 @@ function getPlayerLearnedInstantSpell(cid, name) local p = Player(cid) return p
 function isPlayerGhost(cid) local p = Player(cid) return p and p:isInGhostMode() or false end
 function isPlayerPzLocked(cid) local p = Player(cid) return p and p:isPzLocked() or false end
 function isPremium(cid) local p = Player(cid) return p and p:isPremium() or false end
-function getPlayersByIPAddress(ip, mask)
-	if not mask then mask = 0xFFFFFFFF end
-	local masked = bit.band(ip, mask)
+function getPlayersByIPAddress(ip)
 	local result = {}
 	for _, player in ipairs(Game.getPlayers()) do
-		if bit.band(player:getIp(), mask) == masked then
+		if player:getIp() == masked then
```

Comment:
you have a nil value there, variable `masked` no longer exists within this function scope

Reply:
should look something like this right?
```lua
function getPlayersByIPAddress(ip, mask)
	local players = {}
	if type(id) == "string" then
		for _, player in ipairs(Game.getPlayers()) do
			if player:getIp() == ip then
				players[#players + 1] = player:getId()
			end
		end
	else
		if not mask then mask = 0xFFFFFFFF end
		local masked = bit.band(ip, mask)
		for _, player in ipairs(Game.getPlayers()) do
			if bit.band(player:getIp(), mask) == masked then
				players[#players + 1] = player:getId()
			end
		end
	end
	return players
end
```

Reply:
@xmish yes, that's the reasoning. I just kept this compatibility of accepting numeric IPs for the extremely unlikely, but definitely possible scenario where someone manually blocked an IP by hardcoding it.

Reply:
@Zbizu hope it is good enough now :sweat_smile: